.github/workflows/end-to-end-test.yaml

```diff
@@ -0,0 +1,55 @@
+name: end-to-end-test
+on: push
+jobs:
+  end-to-end:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: checkout the repo
+      uses: actions/checkout@v3
+    - name: build the worker-operator
+      uses: docker/build-push-action@v2
+      with:
+        tags: worker-operator:latest
+        push: false
+    - name: determine node-ip
+      run:  sh .github/workflows/scripts/kind-nodeip.sh
+    - name: create kind clusters
+      run: sh .github/workflows/scripts/kind.sh
+    - name: clone the e2e repo
+      uses: actions/checkout@v3 
```

Comment:
No need to clone the repo and build it here. the docker image can be built in a separate pipeline for e2e repo and push to docker registry. We can just pull the image and use it